homeassistant/components/switchgrid/coordinator.py

```diff
@@ -0,0 +1,50 @@
+"""Coordinator for collecting Switchgrid data."""
+
+from asyncio import timeout
+import logging
+
+import aiohttp
+from switchgrid_python_client import Event, SwitchgridData, SwitchgridEventsResponse
+
+from homeassistant import core
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+import homeassistant.util.dt as dt_util
+
+from .const import UPDATE_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SwitchgridCoordinator(DataUpdateCoordinator[SwitchgridData]):
+    """Coordinator for updating data from the Switchgrid API."""
+
+    def __init__(
+        self,
+        hass: core.HomeAssistant,
+        data: SwitchgridData,
```

Comment:
Can we please rename this to something better? It's a client no?

Reply:
Yup, will rename this to `SwitchgridClient` and update the dependency.